models/migrations/v1_20/v259.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package v1_20 //nolint
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/log"
+	repo_module "code.gitea.io/gitea/modules/repository"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/util"
+	"github.com/google/licensecheck"
+
+	"xorm.io/builder"
+	"xorm.io/xorm"
+)
+
+// Copy paste from models/repo.go because we cannot import models package
+func repoPath(userName, repoName string) string {
+	return filepath.Join(userPath(userName), strings.ToLower(repoName)+".git")
+}
+
+func userPath(userName string) string {
+	return filepath.Join(setting.RepoRootPath, strings.ToLower(userName))
+}
+
+func findLicenseFile(gitRepo *git.Repository, branchName string) (string, *git.TreeEntry, error) {
+	if branchName == "" {
+		return "", nil, nil
+	}
+	if gitRepo == nil {
+		return "", nil, nil
+	}
+
+	commit, err := gitRepo.GetBranchCommit(branchName)
+	if err != nil {
+		if git.IsErrNotExist(err) {
+			return "", nil, nil
+		}
+		return "", nil, fmt.Errorf("GetBranchCommit: %w", err)
+	}
+	entries, err := commit.ListEntries()
+	if err != nil {
+		return "", nil, fmt.Errorf("ListEntries: %w", err)
+	}
+	return repo_module.FindFileInEntries(util.FileTypeLicense, entries, "", "", false)
+}
+
+func detectLicense(file *git.TreeEntry) ([]string, error) {
+	if file == nil {
+		return nil, nil
+	}
+
+	// Read license file content
+	blob := file.Blob()
+	contentBuf, err := blob.GetBlobAll()
+	if err != nil {
+		return nil, fmt.Errorf("GetBlobAll: %w", err)
+	}
+
+	// check license
+	var licenses []string
+	cov := licensecheck.Scan(contentBuf)
+	for _, m := range cov.Match {
+		licenses = append(licenses, m.ID)
+	}
+
+	return licenses, nil
+}
+
+func AddRepositoryLicenses(x *xorm.Engine) error {
+	type Repository struct {
+		ID            int64 `xorm:"pk autoincr"`
+		OwnerName     string
+		Name          string `xorm:"INDEX NOT NULL"`
+		DefaultBranch string
+		Licenses      []string `xorm:"TEXT JSON"`
+	}
+
+	if err := x.Sync(new(Repository)); err != nil {
```

Comment:
![image](https://github.com/go-gitea/gitea/assets/18380374/e21d807f-bdb1-4820-b6a4-c79c9efe61bb)
sync will be failed here.